### DISCUSSION TITLE
CarbonInterface support

### DISCUSSION THREAD

**Discussion by TheAndrey** [spatie/laravel-settings#237] [Ideas]
Hi 👋 
To store the date-time in my settings, I use type `CarbonInterface` as a universal option between `Carbon` and `CarbonImmutable`. In my application, I have configured `Date` facade to use `CarbonImmutable`.

The only problem is that `\Spatie\LaravelSettings\SettingsCasts\DateTimeInterfaceCast` does not understand this type of field and gives an error when reading:

```
Could not cast DateTime type `Carbon\CarbonInterface`
```

I suggest adding type support to `CarbonInterface` by implementing date instantiation through `Date` facade so that the type of object being created depends on the facade `use()` setting.

Now I have to extend the cast class to get what I need:

```php
public function get($payload) : ?\DateTimeInterface {
  if ($payload === null) {
    return null;
  }

  if ($this->type === CarbonInterface::class) {
    return Date::parse($payload);
  }

  return parent::get($payload);
}
```

**Comment by rubenvanassche** [comment]
Hmm, I get your point but I don't think this is the way to go. The settings package is based upon strong types so in this case I would just type CarbonImmutable. When somebody changes the type of the Date facade the type would automatically change too in the settings. Atm there aren't that many differences between the serialization format of Carbon and CarbonImmutable but changing between DateTime and CarbonImmutable would be disastrous.
